src/main/java/com/google/devtools/build/lib/rules/cpp/CppCompileActionBuilder.java

```diff
@@ -361,7 +365,7 @@ NestedSet<Artifact> buildMandatoryInputs() {
     if (grepIncludes != null) {
       realMandatoryInputsBuilder.add(grepIncludes);
     }
-    if (!shouldScanIncludes && dotdFile == null) {
+    if (!shouldScanIncludes && dotdFile == null && !shouldParseShowIncludes()) {
```

Comment:
@oquenchil Do you know why `dotdFile` has to be null here?

Reply:
Those are conditions when graph trimming should not be used. With the compilers supporting dotd files it is when dotd file is not available. For MSVC it is when ParseShowIncludes is turned off. So this condition says "if there is no dotd and also no ParseShowIncludes then unfortunately do not trim the graph"